native/dgesvd.go

```diff
@@ -0,0 +1,1328 @@
+
+						// Copy L to U, zeroing out above it.
+						impl.Dlacpy(blas.Lower, m, m, a, lda, u, ldu)
+						impl.Dlaset(blas.Upper, m-1, m-1, 0, 0, u[1:], ldu)
+
+						ie = itau
+						itauq := ie + m
+						itaup := itauq + m
+						iwork = itaup + m
+
+						// Bidiagonalize L in U.
+						impl.Dgebrd(m, m, u, ldu, s, work[ie:], work[itauq:],
+							work[itaup:], work[iwork:], lwork-iwork)
+
+						// Multiply right bidiagonalizing vectors in U by Q in VT.
+						impl.Dormbr(lapack.ApplyP, blas.Left, blas.Trans, m, n, m,
+							u, ldu, work[itaup:], vt, ldvt, work[iwork:], lwork-iwork)
+
+						// Generate left bidiagonalizing vectors in U.
+						impl.Dorgbr(lapack.ApplyQ, m, m, m, u, ldu, work[itauq:], work[iwork:], lwork-iwork)
+						iwork = ie + m
+
+						// Perform bidiagonal QR iteration, computing left singular
+						// vectors of A in U and computing right singular vectors
+						// of A in VT.
+						ok = impl.Dbdsqr(blas.Upper, m, n, m, 0, s, work[ie:],
+							vt, ldvt, u, ldu, dum, 1, work[iwork:])
+					}
+				}
+			}
+		} else {
+			// Path 10t.
+			// N at least M, but not much larger.
+			ie = 0
+			itauq := ie + m
+			itaup := itauq + m
+			iwork := itaup + m
+
+			// Bidiagonalize A.
+			impl.Dgebrd(m, n, a, lda, s, work[ie:], work[itauq:], work[itaup:], work[iwork:], lwork-iwork)
+			if wantuas {
+				// If left singular vectors desired in U, copy result to U and
+				// generate left bidiagonalizing vectors in U.
+				impl.Dlacpy(blas.Lower, m, m, a, lda, u, ldu)
+				impl.Dorgbr(lapack.ApplyQ, m, m, n, u, ldu, work[itauq:], work[iwork:], lwork-iwork)
+			}
+			if wantvas {
+				// If right singular vectors desired in VT, copy result to VT
+				// and generate right bidiagonalizing vectors in VT.
+				impl.Dlacpy(blas.Upper, m, n, a, lda, vt, ldvt)
+				var nrvt int
+				if wantva {
+					nrvt = n
+				} else {
+					nrvt = m
+				}
+				impl.Dorgbr(lapack.ApplyP, nrvt, n, m, vt, ldvt, work[itaup:], work[iwork:], lwork-iwork)
+			}
+			if wantuo {
+				panic(noSVDO)
+			}
+			if wantvo {
+				panic(noSVDO)
+			}
+			iwork = ie + m
+			var nru, ncvt int
+			if wantuas || wantuo {
+				nru = m
+			}
+			if wantvas || wantvo {
+				ncvt = n
+			}
+			if !wantvo && !wantvo {
+				// Perform bidiagonal QR iteration, if desired, computing left
+				// singular vectors in U and computing right singular vectors in
+				// VT.
+				ok = impl.Dbdsqr(blas.Lower, m, ncvt, nru, 0, s, work[ie:],
+					vt, ldvt, u, ldu, dum, 1, work[iwork:])
+			} else {
+				// There will be two branches when the implementation is complete.
+				panic(noSVDO)
+			}
+		}
+	}
+	if !ok {
+		if ie > 1 {
+			for i := 0; i < minmn-1; i++ {
+				work[i+1] = work[i+ie]
+			}
+		}
+		if ie < 1 {
+			for i := minmn - 2; i >= 0; i-- {
+				work[i+1] = work[i+ie]
+			}
+		}
+	}
+	// Undo scaling if necessary.
+	if iscl {
+		if anrm > bignum {
+			impl.Dlascl(lapack.General, 0, 0, bignum, anrm, minmn, 1, s, minmn)
+		}
+		if ok && anrm > bignum {
```

Comment:
Shouldn't be here `!ok` instead of `ok`? Same below. Fortran has `info.NE.0` and 0 indicates success.